### PR TITLE
docker dist build

### PR DESCRIPTION
**Pull request by tysonnorris**
for #526 
also including:
- a readme.md for docker build + run (local mode)
- docker-compose for running locally

### CODE REVIEW

File: docker/Dockerfile.dist.ubuntu14.04 [hunk 1]
```diff
@@ -0,0 +1,32 @@
+FROM ubuntu:14.04
+
+RUN apt-get update && apt-get -y install \
+      curl \
+      libssl-dev \
+      git \
```
Comment (ajorgensen) [inline docker/Dockerfile.dist.ubuntu14.04]: Do you mind sorting these alphabetically, it helps avoid accidentally adding a dependency twice. 

https://docs.docker.com/engine/userguide/eng-image/dockerfile_best-practices/

**Comment by tysonnorris** [comment]
@ajorgensen good point; I can do the other dockerfiles too, but maybe in a separate PR

File: docker/build-docker.sh [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+cleanup() {
+  if [ -d $SCRATCH_DIR ]; then
+    echo "Cleaning up scratch dir"
+    rm -rf $SCRATCH_DIR
+  fi
+}
+
+trap cleanup EXIT
+
+
```
Comment (ajorgensen) [inline docker/build-docker.sh]: Extra space intentional?

File: docker/build-docker.sh [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+  setup_scratch_dir $SCRATCH_DIR
+
+  echo "building docker image with tag:$DOCKER_TAG"
+  #need to copy artifacts locally
+  cp -pr "$OUTPUT_DIRECTORY"/* "$SCRATCH_DIR/artifacts"
+
+  export HERON_VERSION
+  docker build -t "$DOCKER_TAG" -f "$DOCKER_FILE" "$SCRATCH_DIR"
+
```
Comment (ajorgensen) [inline docker/build-docker.sh]: Extra space intentional?

File: docker/build-docker.sh (hunk 2 above)
Comment (tysonnorris) [inline docker/build-docker.sh]: Yes, I was trying to make this consistent-ish with the other scripts there. If there is a more specific script style I can adjust it.

Comment (ajorgensen) [inline docker/build-docker.sh]: No thats fine, if thats how it is in other scripts then lets keep it consistent.

**Comment by ajorgensen** [comment]
@tysonnorris thats sounds great! I just going to run through it locally and verify that it works but the code changes look good to me.

File: docker/Readme.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+```
+docker/build-docker.sh <platform> <version_string> <output-directory>
+# e.g. docker/build-docker.sh ubuntu14.04 testbuild ~/heron-release
+```
+
+### To run docker containers for local dev work:
+```
+cd docker
+./start-docker.sh heron:<version_string>-<platform>
+# e.g. docker/start-docker.sh heron:testbuild-ubuntu14.04
```
Comment (ajorgensen) [inline docker/Readme.md]: If you cd into the docker directory then this command will not work should be `./start-docker` in your example

File: docker/docker-compose.yml [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+  command: heron-tracker
+  ports:
+    - "8888:8888"
+  volumes:
+    - "herondata:/root/.herondata"
+ui:
+  image: ${HERON_IMAGE}
+  command: heron-ui --tracker_url=http://192.168.33.11:8888
+  ports:
+    - "8889:8889"
```
Comment (ajorgensen) [inline docker/docker-compose.yml]: As a heads up [chef-zero](https://github.com/chef/chef-zero#valid-options) also runs on this port by default and may cause conflicts.

File: docker/docker-compose.yml [hunk 6]
```diff
@@ -0,0 +1,21 @@
+tracker:
+  image: ${HERON_IMAGE}
+  command: heron-tracker
+  ports:
+    - "8888:8888"
+  volumes:
+    - "herondata:/root/.herondata"
+ui:
+  image: ${HERON_IMAGE}
+  command: heron-ui --tracker_url=http://192.168.33.11:8888
```
Comment (ajorgensen) [inline docker/docker-compose.yml]: This ip address will likely not work for anyone else. Is there a better way to link these two containers together without specifying a static ip address?

**Comment by tysonnorris** [comment]
@ajorgensen I cleaned this up a bit, mostly so that the inter-container networking is tidier and the readme is more correct.

**Comment by ajorgensen** [comment]
Looks good, thanks!

**Pull request merged**